src/backend/storage/ipc/ipci.c

```diff
@@ -359,4 +362,45 @@ InitializeShmemGUCs(void)
 		SetConfigOption("shared_memory_size_in_huge_pages", buf,
 						PGC_INTERNAL, PGC_S_DYNAMIC_DEFAULT);
 	}
+
+	/*
+	 * Validating the database server memory limit depends on knowing the size
+	 * of shared memory. This is the first place where we know the size of
+	 * shmem, so it is also the first place we can validate the memory limit.
+	 */
+	if (max_total_memory_mb > 0)
```

Comment:
We had discussed moving this check..?  Just wondering what happened to that change or if there was a reason you decided we should keep this here.